creator-node/src/components/tracks/TranscodeDelegationManager.js

```diff
@@ -0,0 +1,436 @@
+          await TranscodeDelegationManager.handOffToSp(logger, { sp, req })
+
+        decisionTree.state = 'POLLING_FOR_TRANSCODE'
+        const polledTranscodeResponse =
+          await TranscodeDelegationManager.pollForTranscode(logger, {
+            sp,
+            uuid: transcodeAndSegmentUUID
+          })
+
+        decisionTree.state = 'FETCHING_FILES_AND_WRITING_TO_FS'
+        const localFilePaths =
+          await TranscodeDelegationManager.fetchFilesAndWriteToFs(logger, {
+            ...polledTranscodeResponse,
+            sp,
+            fileNameNoExtension: req.fileNameNoExtension
+          })
+
+        resp = localFilePaths
+
+        // If any of these fields are not present, throw and retry onto the next sp
+        if (
+          !resp.transcodeFilePath ||
+          !resp.segmentFileNames ||
+          !resp.m3u8FilePath
+        ) {
+          throw new Error(
+            `Missing fields from transcode fetching response. Actual response=${JSON.stringify(
+              resp
+            )}`
+          )
+        }
+
+        // If the above asserts passed, break out of the loop
+        break
+      } catch (e) {
+        // TODO: delete tmp dir in external SP if fails and continue
+        logger.warn(
+          `Could not hand off track: state=${
+            decisionTree.state
+          } err=${e.toString()}`
+        )
+        console.error(e)
+      }
+    }
+
+    return resp
+  }
+
+  // If any call fails -> throw error
+
+  static async selectRandomSPs(libs) {
+    let allSPs = await libs.ethContracts.getServiceProviderList('content-node')
+    allSPs = allSPs.map((sp) => sp.endpoint)
+
+    // If there are less Content Nodes than the default number, set the cap to the number
+    // of available SPs. This will probably only happen in local dev :shrugs:
+    const numberOfSps =
+      allSPs.length < NUMBER_OF_SPS_FOR_HANDOFF_TRACK
+        ? allSPs.length
+        : NUMBER_OF_SPS_FOR_HANDOFF_TRACK
+
+    const validSPs = new Set()
+    while (validSPs.size < numberOfSps) {
+      const index = Utils.getRandomInt(allSPs.length)
+      const currentSP = allSPs[index]
+      // do not pick self or a node that has already been chosen
+      if (currentSP === CREATOR_NODE_ENDPOINT || validSPs.has(currentSP)) {
+        continue
+      }
+      validSPs.add(currentSP)
+    }
+
+    return Array.from(validSPs)
+  }
+
+  static async handOffToSp(logger, { sp, req }) {
+    const { fileDir, fileName, fileNameNoExtension, uuid: requestID } = req
+
+    await TranscodeDelegationManager.fetchHealthCheck(sp)
+
+    logger.info({ sp }, `Sending off transcode and segmenting request...`)
+    const transcodeAndSegmentUUID =
+      await TranscodeDelegationManager.sendTranscodeAndSegmentRequest({
+        sp,
+        fileDir,
+        fileName,
+        fileNameNoExtension
+      })
+
+    return transcodeAndSegmentUUID
+  }
+
+  static async pollForTranscode(logger, { uuid, sp }) {
```

Comment:
maybe instead of passing in logger to each of these methods (which is a little unintuitive) we can simply pass a logger to `TranscodeDelegationManager.setLogger` or import logger to be used here?